src/engine/source/hlp/src/syntax.hpp

```diff
@@ -0,0 +1,390 @@
+#ifndef _HLP_SYNTAX_HPP
+#define _HLP_SYNTAX_HPP
+
+#include <algorithm>
+#include <stdexcept>
+
+#include "abstractParser.hpp"
+
+/**
+ * @brief Contains the Parser and Result types for the syntax parsers
+ *
+ */
+namespace hlp::syntax
+{
+using ResultT = std::string_view;
+using Result = abs::Result<ResultT>;
+using Parser = abs::Parser<ResultT>;
+
+/**
+ * @brief Given a syntax parser result and the original input, returns the parsed string.
+ *
+ * @param result
+ * @param original
+ * @return std::string_view
+ */
+inline std::string_view parsed(const Result& result, std::string_view original)
+{
+    return original.substr(0, original.size() - result.remaining().size());
+}
+
+/**
+ * @brief Contains the combinators for the syntax parsers, this combinators do not extract any value and do not nest the
+ * results.
+ *
+ */
+namespace combinators
+{
+inline Parser operator&(const Parser& lhs, const Parser& rhs)
```

Comment:
All this overload operators are still of use or should be replaced ? 
e.g. OR (|) by combinator::choice() ?

Reply:
it is used by some syntax parsers